src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AdjudicationsMappingResource.kt

```diff
@@ -309,4 +310,31 @@ class AdjudicationsMappingResource(private val mappingService: AdjudicationMappi
     @PathVariable
     migrationId: String,
   ) = mappingService.deleteAllMappings(migrationId)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_ADJUDICATIONS')")
+  @DeleteMapping("/mapping/adjudications/all")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @Operation(
+    summary = "Delete all adjudication related mapping entries",
+    description = "Delete mapping entries created during any migration or synchronisation for adjudications and associated hearings and punishments",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Adjudication mappings deleted",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  suspend fun deleteAllMappings(
```

Comment:
I'm wondering if separate delete endpoints rather than mutually exclusive request params might be clearer when clearing by type

Reply:
I did consider that, but I think (though not 100% sure yet) we will always want to delete everything, even in live prior to  migration so I wanted an easy default as a priority. And then fiddly flags if you want to do anything different